nuxeo-services/nuxeo-platform-dublincore/src/main/java/org/nuxeo/ecm/platform/dublincore/service/DublinCoreStorageServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+
+    protected static final String DUBLIN_CORE_SCHEMA_NAME = "dublincore";
+
+    public void setCreationDate(DocumentModel doc, Calendar creationDate) {
+        doc.setProperty(DUBLIN_CORE_SCHEMA_NAME, "created", creationDate);
+    }
+
+    public void setIssuedDate(DocumentModel doc, Calendar issuedDate) {
+        doc.setProperty(DUBLIN_CORE_SCHEMA_NAME, "issued", issuedDate);
+    }
+
+    public void setModificationDate(DocumentModel doc, Calendar modificationDate) {
+        doc.setProperty(DUBLIN_CORE_SCHEMA_NAME, "modified", modificationDate);
+        if (doc.getProperty(DUBLIN_CORE_SCHEMA_NAME, "created") == null) {
+            setCreationDate(doc, modificationDate);
+        }
+    }
+
+    public void addContributor(DocumentModel doc, Event event) {
+        Principal principal = event.getContext().getPrincipal();
+        if (principal == null) {
+            return;
+        }
+
+        String principalName = principal.getName();
+        if (principal instanceof SystemPrincipal) {
+            principalName = ((SystemPrincipal) principal).getOriginatingUser();
+            if ((principalName == null || SYSTEM_USERNAME.equals(principalName))
+                    && !ABOUT_TO_CREATE.equals(event.getName())) {
+                return;
+            }
+        }
+
+        if (doc.getProperty(DUBLIN_CORE_SCHEMA_NAME, "creator") == null) {
+            doc.setProperty(DUBLIN_CORE_SCHEMA_NAME, "creator", principalName);
+        }
+
+        List<String> contributorsList = getExistingContributors(doc);
+        if (!contributorsList.contains(principalName)) {
+            contributorsList.add(principalName);
+        }
+        doc.setProperty(DUBLIN_CORE_SCHEMA_NAME, "contributors", contributorsList.toArray());
+        doc.setProperty(DUBLIN_CORE_SCHEMA_NAME, "lastContributor", principalName);
+    }
+
+    private List<String> getExistingContributors(DocumentModel doc) {
+        List<String> contributors = new ArrayList<>();
+        String[] contributorsArray = (String[]) doc.getProperty(DUBLIN_CORE_SCHEMA_NAME, "contributors");
+        if (contributorsArray != null && contributorsArray.length > 0) {
+            Set<String> sanitizedContributors = sanitizeContributors(contributorsArray);
+            contributors.addAll(sanitizedContributors);
+        }
+        return contributors;
+    }
+
+    /**
+     * Returns a "Sanitized" set of contributors according to NXP-25005
+     * @param contributorsArray Unfiltered Contributors Array
+     * @return A set of contributors without repetitions and prefixed entries.
+     */
+    private Set<String> sanitizeContributors(String[] contributorsArray) {
+        Set<String> sanitizedContributors = new LinkedHashSet<>();
+
+        Arrays.stream(contributorsArray).forEach(contributor -> {
+            String unprefixedName = contributor.replace(NuxeoPrincipal.PREFIX, "");
+            sanitizedContributors.add(unprefixedName);
+        });
+
+        return sanitizedContributors;
```

Comment:
We avoid to use a container instantiated outside of lambda because it's generally non necessary and not the philosophy.
Here you migh have something like:
```
Arrays.stream(contributors).map(contributor -> contributor.replace(NuxeoPrincipal.PREFIX, EMPTY)
                           .distinct()
                           .collect(Collectors.toList());
```
You can probably put this code directly in `getExistingContributors`.